WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishedSettingsFragment.kt

```diff
@@ -0,0 +1,95 @@
+package org.wordpress.android.ui.posts
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelProviders
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.ui.posts.EditPostSettingsFragment.EditPostActivityHook
+import javax.inject.Inject
+
+class EditPostPublishedSettingsFragment : Fragment() {
+    private lateinit var dateAndTime: TextView
+    private lateinit var publishNotification: TextView
+    private lateinit var publishNotificationContainer: LinearLayout
+    private lateinit var addToCalendarContainer: LinearLayout
+    @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+    private lateinit var viewModel: EditPostPublishedSettingsViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (activity!!.applicationContext as WordPress).component().inject(this)
+        viewModel = ViewModelProviders.of(activity!!, viewModelFactory)
+                .get(EditPostPublishedSettingsViewModel::class.java)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val rootView = inflater.inflate(R.layout.edit_post_published_settings_fragment, container, false) as ViewGroup
+        dateAndTime = rootView.findViewById(R.id.publish_time_and_date)
+        val dateAndTimeContainer = rootView.findViewById<LinearLayout>(R.id.publish_time_and_date_container)
+        publishNotification = rootView.findViewById(R.id.publish_notification)
+        publishNotificationContainer = rootView.findViewById(R.id.publish_notification_container)
+        addToCalendarContainer = rootView.findViewById(R.id.post_add_to_calendar_container)
+
+        dateAndTimeContainer.setOnClickListener { showPostDateSelectionDialog() }
+
+        viewModel.onDatePicked.observe(this, Observer {
+            showPostTimeSelectionDialog()
```

Comment:
This is called every time on configuration change, even after you picked a time, which results in multiple instances of time picker showing up.

Reply:
great point 👍